docs/distribution/README.md

```diff
@@ -0,0 +1,566 @@
+  docker push $reference
+```
+
+### Generate image manifest and sign it
+
+```shell
+# Generate self-signed certificates
+openssl req \
+  -x509 \
+  -sha256 \
+  -nodes \
+  -newkey rsa:2048 \
+  -days 365 \
+  -subj "/CN=$regIp/O=example inc/C=IN/ST=Haryana/L=Gurgaon" \
+  -addext "subjectAltName=IP:$regIp" \
+  -keyout example.key \
+  -out example.crt
+
+# Generate image manifest
+manifestFile="manifest-to-sign.json" && \
+  docker generate manifest $image > $manifestFile
+
+# Sign manifest
+signatureFile="manifest-signature.jwt" && \
+  nv2 sign --method x509 \
+    -k example.key \
+    -c example.crt \
+    -r $reference \
+    -o $signatureFile \
+    file:$manifestFile
+```
+
+### Obtain manifest and signature digests
+
+```shell
+manifestDigest="sha256:$(sha256sum $manifestFile | cut -d " " -f 1)" && \
+  signatureDigest="sha256:$(sha256sum $signatureFile | cut -d " " -f 1)"
+```
+
+### Create an OCI index file referencing the manifest that was signed and its signature as config
+
+```shell
+indexFile="index.json" && \
+  indexMediaType="application/vnd.oci.image.index.v2+json" && \
+  configMediaType="application/vnd.cncf.notary.config.v2+jwt" && \
+  signatureFileSize=`wc -c < $signatureFile` && \
+  manifestMediaType="$(cat $manifestFile | jq -r '.mediaType')" && \
+  manifestFileSize=`wc -c < $manifestFile`
+
+cat <<EOF > $indexFile
+{
+  "schemaVersion": 2,
+  "mediaType": "$indexMediaType",
+  "config": {
+    "mediaType": "$configMediaType",
+    "digest": "$signatureDigest",
+    "size": $signatureFileSize
+  },
+  "manifests": [
+    {
+      "mediaType": "$manifestMediaType",
+      "digest": "$manifestDigest",
+      "size": $manifestFileSize
+    }
+  ]
+}
+EOF
+```
+
+### Obtain index digest
+
+```shell
+indexDigest="sha256:$(sha256sum $indexFile | cut -d " " -f 1)"
+```
+
+### Push signature and index
+
+```shell
+# Initiate blob upload and obtain PUT location
+configPutLocation=`curl -I -X POST -s http://$registry/v2/$repo/blobs/uploads/ | grep "Location: " | sed -e "s/Location: //;s/$/\&digest=$signatureDigest/;s/\r//"`
+
+# Push signature blob
+curl -X PUT -H "Content-Type: application/octet-stream" --data-binary @"$signatureFile" $configPutLocation
+
+# Push index
+curl -X PUT --data-binary @"$indexFile" -H "Content-Type: $indexMediaType" "http://$registry/v2/$repo/manifests/$indexDigest"
+```
+
+### Retrieve signatures of a manifest as referrer metadata
+
+```shell
+# URL encode index config media type
+metadataMediaType=`python3 -c "import urllib.parse, sys; print(urllib.parse.quote(sys.argv[1]))" $configMediaType`
+
+# Retrieve referrer metadata
+curl -s "http://$registry/v2/$repo/manifests/$manifestDigest/referrer-metadata?media-type=$metadataMediaType" | jq
+```
+
+### Verify signature
+
+```shell
+# Retrieve first signature and store it locally
+metadataDigest=`curl -s "http://$registry/v2/$repo/manifests/$manifestDigest/referrer-metadata?media-type=$metadataMediaType" | jq -r '.referrerMetadata[0]'` && \

```

Comment:
It seems the response is not consistent with the one in the section `referrer-metadata API`